chain/repository.go

```diff
@@ -280,6 +288,31 @@ func (r *Repository) ScanConflicts(blockNum uint32) (uint32, error) {
 	return count, iter.Error()
 }
 
+// ScanHeads returns all head blockIDs from the given blockNum(included) in descending order.
+func (r *Repository) ScanHeads(from uint32) ([]thor.Bytes32, error) {
```

Comment:
I think it'll be more flexible to return kv.Iterator, and simpler code to perform reverse iteration:
```
for ok:= iter.Last(); ok; ok = iter.Prev() {
...
}
```